hack/test-integration.sh

```diff
@@ -42,4 +44,4 @@ if [ -n "${CI:-}" -a -n "${ARTIFACTS:-}" ] ; then
   test_flags="--ginkgo.junit-report=junit.xml"
 fi
 
-GO111MODULE=on go test -timeout=5m -mod=vendor $@ $test_flags | grep -v 'no test files'
+GO111MODULE=on go test -timeout=5m -mod=vendor -ldflags "$LD_FLAGS" $@ $test_flags | grep -v 'no test files'
```

Comment:
> I was unable to reproduce this locally, because running against an actual cluster, the tests won't pass at all because the deployment controller is running and it doesn't allow manually patching the deployment.

In my opinion, this is not the right solution to the problem.
Setting `ldflags` to inject the current version does not help if the corresponding image has not been built yet.
We should rather look into improving the test so that we can run it both against envtest and an existing cluster.
We would need to fake the `Deployment` readiness check in another way.

Reply:
> Setting ldflags to inject the current version does not help if the corresponding image has not been built yet.

Without this, this is defaulted to https://github.com/gardener/gardener/blob/616b60a84b74624228d275d40462ebb22fc6ecb5/vendor/k8s.io/client-go/pkg/version/base.go#L58
So keeping this change doesn't hurt, right?

> We would need to fake the Deployment readiness check in another way.

Do you have something in mind?

Reply:
> So keeping this change doesn't hurt, right?

I'm rather saying it doesn't help to solve any problem, so let's not introduce it.

> Do you have something in mind?

You could fake this on the client side, e.g., by overwriting a function for checking deployment readiness.